### ISSUE TITLE
EasyPHP-DevServer-14.1VC9 Apache stops responding

### ISSUE THREAD

**Issue by pedjas** [easyphp/DEVSERVER#104]
I have freshly installed EasyPHP-DevServer-14.1VC9 on freshly installed Windows 8.1. 

It runs fine, except after some time, not regularly Apache stops responding to http requests.

Sometimes it stops after 10 minutes of starting sometimes it keeps running for few hours and then stops. It does not return any error, just stalls, and web clients are going to timeout.

After I stop and start sever using EasyPHP Systray tool, it starts running again. Using restart option does not help, I have to stop server and then start it again.

I also tried using Apache and MySQL installed as services. It did not help.

**Comment by easyphp** [comment]
Any error message ? What's in the log file ?

**Comment by pedjas** [comment]
On 22.04.2017 03:45, EasyPHP wrote:
> Any error message ? What's in the log file ?

I caught nothing so far. That is the strange part.

**Comment by pedjas** [comment]
On 22.04.2017 03:45, EasyPHP wrote:
> Any error message ? What's in the log file ?

I did simple test:

- Made sure Windows firewall is stopped

- Stopped Apache using EasyPHP

- Deleted all Apache log files

- Started Apache

Apache created few access logs for my development sites (all empty, zero 
bytes), httpd.pid file which contains application PID and error.log 
which contains few lines, usual notices written when server is started, 
no warnings or errors.

Tried to open few test sites I have on this server, everything worked fine.

- I left server up for few hours.

- After few hours when I tried to access site on server, it failed. Web 
browser is just waiting for site response.

- Checked logs:

Access logs are still zero bytes

http.pid file is intact

error.log is intact, it has no more information than what was written in 
that file on server start.

- Stopped Appache server using EasyPHP

At the very moment I stopped server, web browser reported no response 
from http server.

Access log files are changed, access info since server start up until 
stopped is written to those files. No errors in them.

Error log is updated with few notices about shutting down server. No 
errors or warnings.

- Started Apache server

All sites became accessible again.

Only error.log is updated by notices about starting server. No errors 
and no warnings.

**Comment by pedjas** [comment]
Next thing I dod was to raise Apache loggingl evel to debug, and weit to see if something suspicious comes up.

This time it took almost 24 hours for server to halt and this time I did catch some error messages:

[Sun Apr 23 02:49:46.738233 2017] [mpm_winnt:warn] [pid 2976:tid 1260] (OS 64)The specified network name is no longer available.  : AH00341: winnt_accept: Asynchronous AcceptEx failed.

[Sun Apr 23 11:23:09.862760 2017] [mpm_winnt:warn] [pid 2976:tid 1260] (OS 121)The semaphore timeout period has expired.  : AH00341: winnt_accept: Asynchronous AcceptEx failed.

[Sun Apr 23 11:23:09.865760 2017] [mpm_winnt:error] [pid 2976:tid 1260] AH00326: Server ran out of threads to serve requests. Consider raising the ThreadsPerChild setting


Search via Google showed that these are SSL related errors. I do not use SSL at all. I followed some hints about setting httpd.conf so I added these (those did not exist at all in EasyPHP config):

AcceptFilter https none
AcceptFilter http none
EnableSendfile Off
EnableMMAP off

KeepAlive off
KeepAliveTimeout 5
MaxKeepAliveRequests 500


I will wait to see if that helps.
